parquet-column/src/main/java/org/apache/parquet/column/values/bitpacking/ParquetReadRouter.java

```diff
@@ -0,0 +1,132 @@
+  private static final Logger LOG = LoggerFactory.getLogger(ParquetReadRouter.class);
+
+  private static volatile Boolean vector;
+
+  public static void read(int bitWidth, ByteBufferInputStream in, int currentCount, int[] currentBuffer) throws IOException {
+    if (supportVector()) {
+      readBatchVector(bitWidth, in, currentCount, currentBuffer);
+    } else {
+      readBatchVector(bitWidth, in, currentCount, currentBuffer);
+    }
+  }
+
+  public static void readBatchVector(int bitWidth, ByteBufferInputStream in, int currentCount, int[] currentBuffer) throws IOException {
+    BytePacker packer = Packer.LITTLE_ENDIAN.newBytePacker(bitWidth);
+    BytePacker packerVector = Packer.LITTLE_ENDIAN.newBytePackerVector(bitWidth);
+    int valueIndex = 0;
+    int byteIndex = 0;
+    int unpackCount = packerVector.getUnpackCount();
+    int inputByteCountPerVector = packerVector.getUnpackCount() / 8 * bitWidth;
+    int totalByteCount = currentCount * bitWidth / 8;
+
+    // register of avx512 are 512 bits, and can load up to 64 bytes
+    int totalByteCountVector = totalByteCount - 64;
+    ByteBuffer buffer = in.slice(totalByteCount);
+    if (buffer.hasArray()) {
+      for (; byteIndex < totalByteCountVector; byteIndex += inputByteCountPerVector, valueIndex += unpackCount) {
+        packerVector.unpackValuesVector(buffer.array(), buffer.arrayOffset() + buffer.position() + byteIndex, currentBuffer, valueIndex);
+      }
+      // If the remaining bytes size <= 64, the remaining bytes are unpacked by packer
+      for (; byteIndex < totalByteCount; byteIndex += bitWidth, valueIndex += 8) {
+        packer.unpack8Values(buffer.array(), buffer.arrayOffset() + buffer.position() + byteIndex, currentBuffer, valueIndex);
+      }
+    } else {
+      for (; byteIndex < totalByteCountVector; byteIndex += inputByteCountPerVector, valueIndex += unpackCount) {
+        packerVector.unpackValuesVector(buffer, buffer.position() + byteIndex, currentBuffer, valueIndex);
+      }
+      for (; byteIndex < totalByteCount; byteIndex += bitWidth, valueIndex += 8) {
+        packer.unpack8Values(buffer, buffer.position() + byteIndex, currentBuffer, valueIndex);
+      }
+    }
+  }
+  public static void readBatch(int bitWidth, ByteBufferInputStream in, int currentCount, int[] currentBuffer) throws EOFException {
+    BytePacker packer = Packer.LITTLE_ENDIAN.newBytePacker(bitWidth);
+    int valueIndex = 0;
+    while (valueIndex < currentCount) {
+      // values are bit packed 8 at a time, so reading bitWidth will always work
+      ByteBuffer buffer = in.slice(bitWidth);
+      packer.unpack8Values(buffer, buffer.position(), currentBuffer, valueIndex);
+      valueIndex += 8;
+    }
+  }
+
+  public static Boolean supportVector() {
+    if (vector != null) {
+      return vector;
+    }
+    synchronized (ParquetReadRouter.class) {
+      if (vector == null) {
+        synchronized (ParquetReadRouter.class) {
+          vector = avx512Flag();
```

Comment:
I assume the java vector api is agnostic to specific instruction set. So this function name is a little bit misleading to me since it binds the vector to AVX-512. Better to organize the code to be more extensible for other instruction sets.

Reply:
About "the java vector api is agnostic to specific instruction set." @jatin-bhateja Can you help with details ? thanks.
@wgtmac The PR using only avx512 vector to optimize parquet decode. so I have renamed related classes and methods to to be more extensible for other instruction sets(such as AVX2)